src/change/models/changeManager.ts

```diff
@@ -0,0 +1,124 @@
+import { OsmApiElements } from './helpers';
+import { FlattenedGeoJSON, FlattenedGeoJSONPoint, FlattenedGeoJSONLine, FlattenedGeoJSONPolygon } from './geojsonTypes';
+import { ParseOsmElementsError } from './errors';
+import { isNode, isWay } from './helpers';
+
+export const generateOsmChange = (action: Actions, feature: FlattenedGeoJSON, osmElements: OsmApiElements): OsmChange => {
+  switch (feature.geometry.type) {
+    case 'Point': {
+      feature = feature as FlattenedGeoJSONPoint;
+      switch (action) {
+        case Actions.CREATE: {
+          return getChangeFromPoint({ action, feature });
+        }
+        case Actions.MODIFY: {
+          const oldElement = getNodeFromElements(osmElements);
+          return getChangeFromPoint({ action, feature, oldElement });
+        }
+        case Actions.DELETE: {
+          const oldElement = getNodeFromElements(osmElements);
+          return getChangeFromPoint({ action, oldElement });
+        }
+      }
+    }
+    case 'LineString': {
+      feature = feature as FlattenedGeoJSONLine;
+      switch (action) {
+        case Actions.CREATE: {
+          return getChangeFromLine({ action, feature });
+        }
+        case Actions.MODIFY: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromLine({ action, feature, oldElement });
+        }
+        case Actions.DELETE: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromLine({ action, oldElement });
+        }
+      }
+    }
+    case 'Polygon': {
+      feature = feature as FlattenedGeoJSONPolygon;
+      switch (action) {
+        case Actions.CREATE: {
+          return getChangeFromPolygon({ action, feature });
+        }
+        case Actions.MODIFY: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromPolygon({ action, feature, oldElement });
+        }
+        case Actions.DELETE: {
+          const oldElement = getOsmWayFromElements(osmElements);
+          return getChangeFromPolygon({ action, oldElement });
+        }
+      }
+    }
+  }
+};
+
+export const getNodeFromElements = (elements: OsmApiElements): OsmNode => {
+  if (!validateArrayHasElements(elements)) {
+    throwParseOsmElementsError('node');
+  }
+  const node = elements[0];
+  if (!isNode(node)) {
+    throwParseOsmElementsError('node');
+  }
+  return node as OsmNode;
+};
+
+export const getOsmWayFromElements = (elements: OsmApiElements): OsmWay => {
+  const osmWay = parseOsmWayApi(elements);
+  if (osmWay === undefined) {
+    throwParseOsmElementsError('way');
+  }
+  return osmWay as OsmWay;
+};
+
+/**
+ * Gets the tempOsmId generated in the change, used on Create action.
+ * @param {BaseElement[]} elements - Expected to be an array of either one OsmNode or one OsmWay and multiple OsmNodes
+ * @returns {number} The id of either the only existing OsmNode element or OsmWay element
+ */
```

Comment:
The doc is present because of the assumption of the current build of the generated change by [osm-change-generator
](https://github.com/MapColonies/osm-change-generator) thus the function does not validate the data.